dev/com.ibm.ws.config/src/com/ibm/ws/config/admin/internal/ConfigurationStorageHelper.java

```diff
@@ -0,0 +1,673 @@
+        } else if (type == LONG) {
+            value = dis.readLong();
+        } else if (type == FLOAT) {
+            value = dis.readFloat();
+        } else if (type == INTEGER) {
+            value = dis.readInt();
+        } else if (type == STRING) {
+            value = dis.readUTF();
+        } else if (type == PROTECTED_STRING) {
+            value = new SerializableProtectedString(dis.readUTF().toCharArray());
+        } else if (type == ONERROR) {
+            int ordinal = dis.readInt();
+            value = OnError.values()[ordinal];
+        } else {
+            value = null;
+        }
+
+        return value;
+    }
+
+    private static Collection<?> readCollection(DataInputStream dis) throws IOException {
+        List<Object> result = new ArrayList<>();
+
+        int len = dis.readInt();
+        if (len > 0) {
+            int type = dis.readByte();
+            for (int i = 0; i < len; i++) {
+                result.add(readPrimative(dis, type));
+            }
+        }
+        return result;
+    }
+
+    public static void writeMap(DataOutputStream dos, MapIterable map) throws IOException {
+        if (map == null) {
+            dos.writeInt(-1);
+            return;
+        }
+        dos.writeInt(map.size());
+        for (Map.Entry<String, Object> entry : map) {
+            dos.writeUTF(entry.getKey());
+            Object obj = entry.getValue();
+            if (obj == null) {
+                dos.writeByte(NULL);
+            } else if (obj instanceof Collection) {
+                dos.writeByte(COLLECTION);
+                Collection<?> data = (Collection<?>) obj;
+                dos.writeInt(data.size());
+                boolean writeType = true;
+                for (Object colObj : data) {
+                    // write primative.
+                    if (writeType) {
+                        writePrimitiveType(dos, colObj.getClass());
+                        writeType = false;
+                    }
+                    writePrimitive(dos, colObj);
+                }
+            } else if (obj instanceof Map) {
+                dos.writeByte(MAP);
+                writeMap(dos, toMapOrDictionary(obj));
+            } else if (obj.getClass().isArray()) {
+                dos.writeByte(ARRAY);
+                Class<?> arrayType = obj.getClass();
+                arrayType = arrayType.getComponentType();
+                writePrimitiveType(dos, arrayType);
+                dos.writeBoolean(arrayType.isPrimitive());
+                int len = Array.getLength(obj);
+                dos.writeInt(len);
+                for (int i = 0; i < len; i++) {
+                    writePrimitive(dos, Array.get(obj, i));
+                }
+            } else {
+                writePrimitiveType(dos, obj.getClass());
+                writePrimitive(dos, obj);
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public static MapIterable toMapOrDictionary(Object obj) {
```

Comment:
does this need to be public?

Reply:
Yes for usage by the variable registry cache